src/pkg/caendr/caendr/utils/tokens.py

```diff
@@ -75,6 +76,25 @@ def __add__(self, other):
       raise ValueError(f'Cannot replace tokens in non-string value {other}')
 
 
+  def apply(f, *args, **kwargs):
```

Comment:
As mentioned above, the `TokenizedString` class ended up a bit complicated without great justification, so I'd like to streamline it when we have more time.  In the meantime, this method handles mapping a string-based function to a set of `TokenizedString` arguments, dealing w the tokens implicitly.